### PR TITLE
fix(docs): Update react guide

### PR DESCRIPTION
**Pull request by mister-ben**
Update react doc to address some issues that crept in when the docs were moved

Fixes videojs/video.js#7780

### CODE REVIEW

**Comment by gkatsev** [comment]
I'm a bit surprised that this change is needed. Why would the player reference not be available as currently done?

**Comment by gkatsev** [comment]
Actually, I see some of the items. I think for the logs we can still continue using `player.log`?

File: src/mdx-pages/guides/react.mdx [hunk 1]
```diff
@@ -28,14 +28,17 @@ export const VideoJS = (props) => {
...
-    // You can update player in the `else` block here, for example:
+    // You could update an existing player in the `else` block here
+    // on prop change, for example:
     } else {
-      player.autoplay(options.autoplay);
-      player.src(options.sources);
+      // const player = playerRef.current;
+
+      // player.autoplay(options.autoplay);
+      // player.src(options.sources);
```
Comment (gkatsev) [inline src/mdx-pages/guides/react.mdx:41]: I believe this is the big fix in this PR. Any reason to use `videojs.log` over `player.log`, it seems like that should work.

Comment (mister-ben) [reply src/mdx-pages/guides/react.mdx:41]: `player` isn't defined everywhere it was used, so just made it `videojs.log` to keep it simple.

Comment (gkatsev) [reply src/mdx-pages/guides/react.mdx:41]: isn't it defined on line 30?

Comment (gkatsev) [reply src/mdx-pages/guides/react.mdx:41]: oh, I see, sorry I was conflating the examples into a single one. Yeah, I guess it's fine, it isn't a huge difference either way.

**Review by gkatsev — APPROVED** [review]

**Review by misteroneill — APPROVED** [review]

**Pull request merged**